core/src/main/java/com/oath/oak/OakRKeyBuffer.java

```diff
@@ -130,8 +130,8 @@ private ByteBuffer getTemporaryPerThreadByteBuffer() {
     @Override
     public ByteBuffer getByteBuffer() {
         ByteBuffer buff = getTemporaryPerThreadByteBuffer().asReadOnlyBuffer();
-        buff.position(headerSize + position);
-        buff.limit(headerSize + position + length);
+        buff.position(position + headerSize);
+        buff.limit(position + length);
```

Comment:
1) I wonder why do we need to set the limit here if Block's getBufferForThread() method already does this? To set position not to include the headerSize is enough.
2) asReadOnlyBuffer() already creates a Buffer object as does the slice() in the last line. Can we avoid 2 objects instead of 1?
3) Also please add a note that length includes the header size (this is true at least for value)

Reply:
1. We have to set the position, but the limit is redundant. I replaced it with a comment.
2. Unfortunately, no. ByteBuffer implementation does not support that.
3. Done.